### PR TITLE
feat(backup): Update to allow use of backup exclusion file for omitting content from backups

### PR DESCRIPTION
**Pull request by jimmyatSplunk**
I've reworked the backup command to use a flat file that is used for file and folder exclusions in an attempt to allow admins to exclude files or folders from their backups. The idea being they can exclude game files that will be replaced by repairing the game server and it helps keep their backups smaller.

I've also added the necessary exclude file variable and a check for the "backupexclude.cfg" in the same fashion as "common.cfg" and "instance.cfg" so it should fold into current installations without issue.

### CODE REVIEW

**Comment by jimmyatSplunk** [comment]
This request is essentially the same request as https://github.com/GameServerManagers/LinuxGSM/pull/1595#issue-137964814 but there are more changes in that that have nothing to do with implementing this improvement so it should be rejected and closed.

**Comment by Frisasky** [comment]
please commit into develop branch as master branch are used for script release

**Comment by dgibbs64** [comment]
Looks great :). I will need to review this to make sure I fully understand how this all works and do testing. So please bear with me on this might take a little time to merge

**Comment by dgibbs64** [comment]
I think a backup re-factor is in order

File: linuxgsm.sh [hunk 1]
```diff
@@ -20,7 +20,7 @@ if [ -f ".dev-debug" ]; then
 	set -x
 fi
 
-version="190301"
+version="190422"
```
Comment (h3o66) [inline linuxgsm.sh]: i guess this should not be in the commit, as the version is changed by release.

**Review by h3o66 — CHANGES_REQUESTED** [review]
Please rebase and resolve the conflicts

**Pull request closed without merging**